File: src/lecture1/lecture1_my_topic_cpp/src/publisher.cpp
#include <ros/ros.h>
#include <lecture1_msgs/TimeUInt32.h>

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "publisher");
    ros::NodeHandle n;
    ros::Publisher p = n.advertise<lecture1_msgs::TimeUInt32>("topic", 5);
    ros::Rate rate(10);
    int c = 0;
    while (ros::ok())
    {
        lecture1_msgs::TimeUInt32 msg;
        msg.stamp = ros::Time::now();
        msg.data = c++;
        p.publish(msg);
        rate.sleep();
    }
    return 0;
}

File: src/lecture1/lecture1_my_topic_py/scripts/publisher.py
#!/usr/bin/env python3
import rospy
from lecture1_msgs.msg import TimeUInt32

def publisher():
    rospy.init_node("publisher")
    p = rospy.Publisher("topic", TimeUInt32, queue_size=5)
    rate = rospy.Rate(10)
    c = 0
    while not rospy.is_shutdown():
        p.publish(rospy.Time.now(), c)
        c += 1
        rate.sleep()

if __name__ == '__main__':
    try:
        publisher()
    except rospy.ROSInterruptException:
        pass


